### ISSUE TITLE
Update all subscribe calls to use recommended consumption pattern

### ISSUE THREAD

**Issue by faibz** [smp-org/smp#55]
Currently, our subscribe calls are structured as follows:
```
x.subscribe(
(...) => {
   //success
},
error => {
   //fail
});
```
However, this style is deprecated in favour of using observers. This issue is to update our subscribe calls to use the recommended consumption pattern.

**Comment by faibz** [comment]
Based on this comment https://stackoverflow.com/a/55472361/10647081 we should update our subscribes to look like this:
`
x.subscribe({
    next: (...) => {
        //success
    },
    error: error => {
        //fail
    }
});
`

**Comment by faibz** [comment]
Fixed in #57
